addons/web_editor/static/src/js/editor/odoo-editor/src/OdooEditor.js

```diff
@@ -3626,9 +3626,11 @@ export class OdooEditor extends EventTarget {
             const ancestorsList = [commonAncestorElement, ...ancestors(commonAncestorElement, blockEl)];
             // Wrap rangeContent with clones of their ancestors to keep the styles.
             for (const ancestor of ancestorsList) {
-                const clone = ancestor.cloneNode();
-                clone.append(...rangeContent.childNodes);
-                rangeContent.appendChild(clone);
+                if (ancestor.nodeName !== 'DIV') {
```

Comment:
It's a bit arbitrary to just not include `div`s specifically. I see that you had introduced this code so that parent formats (b, em, span, ...) would be included. I think the real issue here is that we're wrapping the content in _block_ ancestors: while you do stop at `blockEl`, `commonAncestorElement` might itself be a block. If I'm not mistaken, we only want to wrap inline nodes in all their inline ancestors.

Reply:
No, we need to wrap the block node too. otherwise block elements like `h1`, `pre` won't be copied.

Reply:
Then wrap only non-blocks or nodes in `paragraphRelatedElements`.

Reply:
I have implemented the requested changes.